### PR TITLE
Enable multiple file extensions

### PR DESCRIPTION
**Pull request by akaIDIOT**
References #54, detect and load `.yml` by default.

### CODE REVIEW

**Comment by HHoofs** [comment]
Nice

File: confidence/io.py [hunk 1]
```diff
@@ -270,7 +270,7 @@ def loads(*strings, missing=Missing.silent):
     return Configuration(*(yaml.safe_load(string) for string in strings), missing=missing)
 
 
-def load_name(*names, load_order=DEFAULT_LOAD_ORDER, extension='yaml', missing=Missing.silent):
+def load_name(*names, load_order=DEFAULT_LOAD_ORDER, extension=('yaml', 'yml'), missing=Missing.silent):
```
Comment (akaIDIOT) [inline confidence/io.py:273]: Maybe this should not be the default. Anyone encountering `yml` files can enable it this way or even hardcode it to just `yml`. The doc states this parameter can be either a `str` or a sequence, should the singular default be a `str` or a one-value `tuple`? :thinking:

Comment (akaIDIOT) [reply confidence/io.py]: FXIED, defaulting to just plain old `yaml`, turning this PR into an 'it is now possible to ...' :)

**Comment by HHoofs** [comment]
What if I have `42.yaml` and `42.yml` on the same 'level'?

**Comment by akaIDIOT** [comment]
> What if I have `42.yaml` and `42.yml` on the same 'level'?

The use of `itertools.product` would read them both, in a deterministic order (and hence precedence). Whether that was meant to happen is another question. We can simply state that that's a problem for the user, but maybe we should involve a user in the 'expected behaviour' discussion? :) What would you expect to happen?

**Comment by HHoofs** [comment]
I would expect to have an error 😅
It seems like the user has two different configurations that should be treated equally. I would want to know that I'm doing something stupid...

**Comment by akaIDIOT** [comment]
After some discussion, this PR maybe catering to a use case we'd want to discourage and not support like this. Users mixing `.yml` and `.yaml` files can load both and combine them in any way they see fit, confidence should maybe not do this out of the box. Agreed? :sweat_smile:

**Comment by HHoofs** [comment]
Totally agree, I cannot think of a use case that warrants such iffy behaviour

**Comment by akaIDIOT** [comment]
On second thought, let's not go to PR, 'tis a silly place.

Closes #54

**Pull request closed without merging**